OpenRA.Game/Map/MapCache.cs

```diff
@@ -250,7 +250,20 @@ IEnumerator IEnumerable.GetEnumerator()
 
 		public void Dispose()
 		{
-			sheetBuilder.Dispose();
+			if (previewLoaderThread == null)
+			{
+				sheetBuilder.Dispose();
+				return;
+			}
+
+			// We need to let the loader thread exit before we can dispose our sheet builder.
+			// Ideally we should dispose our resources before returning, but we don't to block waiting on the loader thread to exit.
+			// Instead, we'll queue disposal to be run once it has exited.
+			ThreadPool.QueueUserWorkItem(_ =>
```

Comment:
All our other code that needs a worker will explicitly spawn a new thread.  Do we want to convert those to use ThreadPool in a future PR?